src/main/java/com/google/sps/servlets/likesServlet.java

```diff
@@ -53,22 +53,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     likesEntity.setProperty("userId", userId_like);
     likesEntity.setProperty("postId", postId_like);
 	
-    Query query_likes = new Query("Likes");
-    PreparedQuery results_likes = datastore.prepare(query_likes);
-	
-    /** Check for multiple likes by a user for the same comment */
     bool same_entry = false;
-    for (Entity entity : results_likes.asIterable()) {
-      String userId = (String) entity.getProperty("userId");
-      long postId = (long) entity.getProperty("postId");
-      if(userId==userId_like && postId==postId_like){
-        same_entry = true;
-	break;
-      }
-    }
-    if(!same_entry){
+
+    /** Check for same entry and then add to datastore*/
+    Query<Entity> query_likes = Query.newEntityQueryBuilder().setKind("Likes").setFilter(PropertyFilter.eq("userId", userId_like), PropertyFilter.eq("postId", postId_like)).build();
```

Comment:
`
query_like = Query.newEntityQuweryBuilder()
      .setKind()
      .setFilter()
`
The above will make your code more readable

Reply:
Resolved